src/Features/VisualBasic/Portable/Diagnostics/Analyzers/VisualBasicRemoveUnnecessaryCastDiagnosticAnalyzer.vb

```diff
@@ -11,19 +11,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Diagnostics.RemoveUnnecessaryCast
 
     <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
     Friend NotInheritable Class VisualBasicRemoveUnnecessaryCastDiagnosticAnalyzer
-        Inherits RemoveUnnecessaryCastDiagnosticAnalyzerBase(Of SyntaxKind)
+        Inherits RemoveUnnecessaryCastDiagnosticAnalyzerBase(Of SyntaxKind, ExpressionSyntax)
 
-        Private Shared ReadOnly s_kindsOfInterest As ImmutableArray(Of SyntaxKind) = ImmutableArray.Create(SyntaxKind.CTypeExpression,
-                                                                                                          SyntaxKind.DirectCastExpression,
-                                                                                                          SyntaxKind.TryCastExpression,
-                                                                                                          SyntaxKind.PredefinedCastExpression)
-        Public Overrides ReadOnly Property SyntaxKindsOfInterest As ImmutableArray(Of SyntaxKind)
-            Get
-                Return s_kindsOfInterest
-            End Get
-        End Property
+        Protected Overrides ReadOnly Property SyntaxKindsOfInterest As ImmutableArray(Of SyntaxKind) =
+            ImmutableArray.Create(SyntaxKind.CTypeExpression,
+                                  SyntaxKind.DirectCastExpression,
+                                  SyntaxKind.TryCastExpression,
+                                  SyntaxKind.PredefinedCastExpression)
 
-        Protected Overrides Function IsUnnecessaryCast(model As SemanticModel, node As SyntaxNode, cancellationToken As CancellationToken) As Boolean
+        Protected Overrides Function IsUnnecessaryCast(model As SemanticModel, node As ExpressionSyntax, cancellationToken As CancellationToken) As Boolean
             Select Case node.Kind
                 Case SyntaxKind.CTypeExpression, SyntaxKind.DirectCastExpression, SyntaxKind.TryCastExpression
                     Return DirectCast(node, CastExpressionSyntax).IsUnnecessaryCast(model, assumeCallKeyword:=True, cancellationToken:=cancellationToken)
```

Comment:
we probably need to build IsUnnecessaryCast kind of helper for IOperation. or create helper that internally get syntax node and use existing one, but looks like helper on IOperation.